app/commands/v2/patch_link_set.rb

```diff
@@ -25,20 +24,24 @@ def call
         # we need to reload the link_set as the links association will be stale
         link_set.reload
 
-        orphaned_content_ids = link_diff_between(before_links.map(&:target_content_id), link_set.links.map(&:target_content_id))
+        orphaned_content_ids = link_diff_between(@before_links.map(&:target_content_id), link_set.links.map(&:target_content_id))
 
         after_transaction_commit do
           send_downstream(orphaned_content_ids)
         end
 
-        Action.create_patch_link_set_action(link_set, before_links, event)
+        Action.create_patch_link_set_action(link_set, @before_links, event)
 
         presented = Presenters::Queries::LinkSetPresenter.present(link_set)
         Success.new(presented)
       end
 
     private
 
+      def link_set
+        @link_set ||= LinkSet.find_or_create_locked(content_id: content_id)
+      end
```

Comment:
I'm reluctantly OK with the pattern of functions like this for queries, but given this can actually change the database, I'm unsure what this means for the readability of the code.

In my mind, it would be easier to see where the find or create happens if in `call` there was
```ruby
@link_set = LinkSet.find_or_create_locked(content_id: content_id)
```

Reply:
Yeah, I think you're right. It makes it harder to know exactly at which point the potential create happens. I've amended the commit without this method.

Reply:
Great, that's looking really good. I like making `update_dependencies` an argument as well, as that makes it clearer to see where it's used also. :+1: